clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1581,7 +1581,28 @@ public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, Of
      * Overrides the fetch offsets that the consumer will use on the next {@link #poll(Duration) poll(timeout)}. If this API
      * is invoked for the same partition more than once, the latest offset will be used on the next poll(). Note that
      * you may lose data if this API is arbitrarily used in the middle of consumption, to reset the fetch offsets
+     * <p>
+     * The next Consumer Record which will be retrieved when poll() is invoked will have the offset specified, given that
+     * a record with that offset exists (i.e. it is a valid offset).
+     * <p>
+     * {@link #seekToBeginning(Collection)} will go to the first offset in the topic.
+     * seek(0) is equivalent to seekToBeginning for a TopicPartition with beginning offset 0,
+     * assuming that there is a record at offset 0 still available.
+     * {@link #seekToEnd(Collection)} is equivalent to seeking to the last offset of the partition, but behavior depends on
+     * {@code isolation.level}, so see {@link #seekToBeginning(Collection)} documentation for more details.
+     * <p>
+     * Seeking to the offset smaller than the log start offset or larger than the log end offset
+     * or high watermark means an invalid offset is reached.
+     * Invalid offset behaviour is controlled by the {@link ConsumerConfig AUTO_RESET_CONFIG} property.
+     * If this is set to "earliest", the next poll will return records from the starting offset.
+     * If it is set to "latest", it will seek to the last offset (similar to seekToEnd()).
+     * If it is set to "none", an {@code OffsetOutOfRangeException} will be thrown.
+     * <p>
+     * Note that, the seek offset won't change to the in-flight fetch request, it will take effect in next fetch request.
+     * So, the consumer might wait for {@code fetch.max.wait.ms} before starting to fetch the records from desired offset.
      *
+     * @param partition The TopicPartition on which the seek will be performed.
+     * @param offset the next offset returned by poll() will be either this or greater.
```

Comment:
About the `greater`, I don't know what you are referring to. Could you elaborate more? 
I think it's possible we returned greater offset if this is a compact topic. But I don't think we need to make this line that complicated.

Reply:
i think i had seen an odd topic on a project which had a custom kafka topic viewer where the offsets were all even numbers (or maybe multiples of 4) for a sequence of messages, but that might have just been a ui bug; will leave it out for now

Reply:
the main point is if you seek to offset 5, you'll also get offset 5 back (as opposed to the next message being 6)